src/object/cli_obj.c

```diff
@@ -2501,6 +2560,14 @@ do_dc_obj_fetch(tse_task_t *task, daos_obj_fetch_t *args,
 	if (rc != 0)
 		D_GOTO(out_task, rc);
 
+	if (!obj_auxi->io_retry) {
+		rc = csum_obj_fetch(obj, args, obj_auxi);
```

Comment:
minor, maybe can return if !io_retry inside csum_obj_fetch to be the same style as obj_rw_bulk_prep. same for csum_obj_update